src/compiler/tsc.ts

```diff
@@ -360,7 +360,7 @@ namespace ts {
             let newFileNames = ts.map(parsedCommandLine.fileNames, compilerHost.getCanonicalFileName);
             let canonicalRootFileNames = ts.map(rootFileNames, compilerHost.getCanonicalFileName);
 
-            if (!arrayStructurallyIsEqualTo(newFileNames, canonicalRootFileNames)) {
+            if (!arrayIsEqualTo(newFileNames.sort(), canonicalRootFileNames.sort())) {

```

Comment:
Considering @vladima's comment and the fact that structrual comparison would probably be used in other places later, it might be better to change it back to a generic option inside the `arrayIsEqualTo`. Only difference is I'll add an optional comparer and don't do the sort in-place.